recipes/sol2/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy
+
+required_conan_version = ">=1.53.0"
+
+
+class Sol2Conan(ConanFile):
+    name = "sol2"
+    package_type = "header-library"
+    description = "a C++ <-> Lua API wrapper with advanced features and top notch performance"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ThePhD/sol2"
+    topics = ("lua", "c++", "bindings", "scripting")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    options = {
+        "with_lua": ["lua", "luajit"]
+    }
+
+    default_options = {
+        "with_lua": "lua"
+    }
+
+    def requirements(self):
+        if self.options.with_lua == "lua":
+            if self.version < "3.1.0":
+                # v2.x.x & v3.0.x supports up to Lua 5.3
+                self.requires("lua/5.3.6")
+            else:
+                self.requires("lua/5.4.4")
+        elif self.options.with_lua == "luajit":
+            self.requires("luajit/2.1.0-beta3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    @property
+    def _compilers_minimum_version(self):
+        # TODO: MSVC?
+        if self.version < "3.0.0":
+            return {
+                "Visual Studio": "14",
+                "gcc": "5",
+                "clang": "3.2",
+                "apple-clang": "4.3"
+            }
+        else:
+            return {
+                "Visual Studio": "15.7" if Version(self.version) < "3.3.0" else "16",
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "10",
+            }
+
+    @property
+    def _min_cppstd(self):
+        if self.version < "3.0.0":
```

Comment:
I am actually not sure what the answer is on this one... but putting it the team lol

```suggestion
        if Version(self.version) < "3.0.0":
```

Reply:
Yes, explicitly converting to version would be better. The ``conan.tools.scm`` ``Version`` does a ``str()`` internally, so even if ``self.version`` is a Version in 1.X, it should be safe.